lib/parsers/exif_parser.rb

```diff
@@ -65,8 +52,104 @@ def to_json(*maybe_coder)
     end
 
     def orientation
-      value = __getobj__.orientation.to_i
-      ORIENTATIONS.fetch(value - 1)
+      __getobj__.orientation.to_i
+    end
+
+    def orientation_sym
+      ORIENTATIONS.fetch(orientation)
+    end
+  end
+
+  # With some formats, multiple EXIF tag frames can be included in a single file.
+  # For example, JPEGs might have multiple APP1 markers which each contain EXIF
+  # data. The EXIF data in them, however, is not necessarily "complete" - it seems
+  # most applications assume that these blocks "overwrite" each other with the properties
+  # they specify. Probably this is done for more efficient saving - instead of overwriting
+  # the EXIF data with a modified version - which would also potentially disturb any digital
+  # signing that this data might include - the applications are supposed to follow the order
+  # in which these tags appear in the file:
+  #
+  # Take a resized image for example:
+  #
+  #   APP1 {author: 'John', pixel_width: 1024}
+  #   APP1 {pixel_width: 10}
+  #
+  # That image would get a combined EXIF of:
+  #
+  #   APP1 {author: 'John', pixel_width: 10}
+  #
+  # since the frame that comes later in the file overwrites a property. From what I see
+  # exiftools do this is the way it works.
+  #
+  # This class acts as a wrapper for this "layering" of chunks of EXIF properties, and will
+  # follow the following conventions:
+  #
+  # * When merging data for JSON conversion, it will merge it top-down. It will overwrite
+  #   any specified properties. An exception is made for orientation (see below)
+  # * When retrieving a property, it will look "from the end to the beginning" of the EXIF
+  #   dataframe stack, looking for the first dataframe which has this property with a non-nil value
+  # * When retrieving orientation, it will pick the first orientation value which is not nil
```

Comment:
If I understand this correctly a JPG can have two (or more) layers of orientation information both telling us something different?

What is reasoning we use different logic for the orientation information vs the other EXIF data?

Reply:
The reason is that exifr returns the same result (`0`) both when the `0` is recorded in the EXIF tag _and_ when it is simply not present. We need to make a distinction between the two and fallback to the EXIF tag "above" in the latter.